### PR TITLE
[api] Add new /rename public api

### PR DESCRIPTION
**Pull request by nidhibhatg**
## What changes were proposed in this pull request?

- Implement a dedicated /rename API method tailored for new storage browser.
- This separates the old method for existing filebrowser to not have regressions + much cleaner new method.

## How was this patch tested?

- Manually

Please review [Hue Contributing Guide](https://github.com/cloudera/hue/blob/master/CONTRIBUTING.md) before opening a pull request.

### CODE REVIEW

File: apps/filebrowser/src/filebrowser/api.py [hunk 1]
```diff
@@ -108,3 +109,22 @@ def touch(request):
   
   request.fs.create(request.fs.join(path, name))
   return HttpResponse(status=200)
+
+@error_handler
+def rename(request):
+  src_path = request.POST.get('src_path')
+  dest_path = request.POST.get('dest_path')
+
+  """If dest_path doesn't have a directory specified, use same dir."""
```
Comment (Harshg999) [inline apps/filebrowser/src/filebrowser/api.py:118]: Why is this required?

File: apps/filebrowser/src/filebrowser/api.py [hunk 2]
```diff
@@ -108,3 +109,22 @@ def touch(request):
...
+  if "#" in dest_path:
+    raise Exception(_(
+      "Error renaming %s to %s. Hashes are not allowed in file or directory names." % (os.path.basename(src_path), dest_path)
+      ))
+  if "/" not in dest_path:
```
Comment (Harshg999) [inline apps/filebrowser/src/filebrowser/api.py:123]: Nit: Space

File: apps/filebrowser/src/filebrowser/api.py [hunk 3]
```diff
@@ -108,3 +109,22 @@ def touch(request):
...
+    src_dir = os.path.dirname(src_path)
+    dest_path = request.fs.join(src_dir, dest_path)
+  if request.fs.exists(dest_path):
```
Comment (Harshg999) [inline apps/filebrowser/src/filebrowser/api.py:126]: Nit: Space

**Review by Harshg999 — APPROVED** [review]
LGTM!

File: apps/filebrowser/src/filebrowser/api.py (hunk 1 above)
Comment (nidhibhatg) [reply apps/filebrowser/src/filebrowser/api.py:118]: Rename api works in both ways :
Ex: renaming user/test to user/test123 
1) You can specify the whole path for the dest_path ie dest_path = user/test123 
2) Usually when the api is called from UI dest_path only consists of the new name for the file or folder with no directory specified ie dest_path = test123. Since a directory isnt specified in this scenario it uses the current directory path when calling request.fs.rename() function.

Comment (Harshg999) [reply apps/filebrowser/src/filebrowser/api.py:118]: I see, let's add it as a comment before https://github.com/cloudera/hue/pull/3619/files#diff-d8a7a37b125cd146d72b25ec4c75d6c208194ebc09ab376f0445cabc76af1bb1R119?

**Review by bjornalm — APPROVED** [review]
LGTM
